File: ssh-manager/src/app/composables/tauri/commands/profiles.ts
import { invoke } from '@tauri-apps/api/core'
import type { AppProfile } from '~/types'

export async function getAppProfiles(): Promise<AppProfile[]> {
  return invoke<AppProfile[]>('get_app_profiles')
}

export async function saveAppProfile(profile: AppProfile): Promise<void> {
  return invoke('save_app_profile', { profile })
}

export async function deleteAppProfile(id: string): Promise<void> {
  return invoke('delete_app_profile', { id })
}

File: ssh-manager/src-tauri/src/commands/profiles.rs
use tauri::State;

use crate::config_store::ConfigStore;
use crate::models::AppProfile;

#[tauri::command]
pub fn get_app_profiles(store: State<'_, ConfigStore>) -> Result<Vec<AppProfile>, String> {
    store.get_all_profiles()
}

#[tauri::command]
pub fn save_app_profile(store: State<'_, ConfigStore>, profile: AppProfile) -> Result<(), String> {
    store.save_profile(&profile)
}

#[tauri::command]
pub fn delete_app_profile(store: State<'_, ConfigStore>, id: String) -> Result<(), String> {
    store.delete_profile(&id)
}
